modules/ballerina-core/src/main/java/org/ballerinalang/helpers/DeepEqualHelper.java

```diff
@@ -0,0 +1,279 @@
+                    return false;
+                }
+                break;
+            case TypeTags.ENUM_TAG:
+                BEnumerator lhEnum = (BEnumerator) lhVal;
+                BEnumerator rhEnum = (BEnumerator) rhVal;
+                
+                if (!lhEnum.stringValue().equals(rhEnum.stringValue())) {
+                    return false;
+                }
+                break;
+            default:
+                return false;
+        }
+        
+        return true;
+    }
+    
+    /**
+     * Deep equals function for JSON type values. I.E ===
+     * @param lhJson The left side value.
+     * @param rhJson The right side value.
+     * @return True if values are equal, else false.
+     */
+    public static boolean isDeepEqual(JsonNode lhJson, JsonNode rhJson) {
+        if (lhJson.getType() == JsonNode.Type.OBJECT) {
+            // Converting iterators to maps as iterators are ordered.
+            Iterator<Map.Entry<String, JsonNode>> lhJsonFieldsIterator = lhJson.fields();
+            HashMap<String, JsonNode> lhJsonFields = new HashMap<>();
+            while (lhJsonFieldsIterator.hasNext()) {
+                Map.Entry<String, JsonNode> field = lhJsonFieldsIterator.next();
+                lhJsonFields.put(field.getKey(), field.getValue());
+            }
+    
+            Iterator<Map.Entry<String, JsonNode>> rhJsonFieldsIterator = rhJson.fields();
+            HashMap<String, JsonNode> rhJsonFields = new HashMap<>();
+            while (rhJsonFieldsIterator.hasNext()) {
+                Map.Entry<String, JsonNode> field = rhJsonFieldsIterator.next();
+                rhJsonFields.put(field.getKey(), field.getValue());
+            }
+            
+            // Size of the fields should be same.
+            if (lhJsonFields.size() != rhJsonFields.size()) {
+                return false;
+            }
+    
+            if (lhJsonFields.size() > 0) {
+                for (String key : lhJsonFields.keySet()) {
+                    if (!rhJsonFields.keySet().contains(key)) {
+                        return false;
+                    }
+                    if (!isDeepEqual(lhJsonFields.get(key), rhJsonFields.get(key))) {
+                        return false;
+                    }
+                }
+            }
+        }
+        
+        if (lhJson.getType() == JsonNode.Type.ARRAY) {
+            if (lhJson.size() != rhJson.size()) {
+                return false;
+            }
+            
+            if (lhJson.size() > 0) {
+                for (int i = 0; i < lhJson.size(); i++) {
+                    if (!isDeepEqual(lhJson.get(i), rhJson.get(i))) {
+                        return false;
+                    }
+                }
+            }
+        }
+        
+        // If it is a value type, then equalize their text values.
+        if (!lhJson.asText().equals(rhJson.asText())) {
```

Comment:
We can remove this if statement and directly have `return lhJson.asText().equals(rhJson.asText();` here

Reply:
Added direct return statement. - https://github.com/ballerina-lang/ballerina/pull/4532/commits/c728619b2fcdb290f1587e71e11911955ccfae3f